### PR TITLE
Prepare to release to 0.22.1

### PR DESCRIPTION
**Pull request by tomhjp**
Updates changelog and does some version updates:
* Vault
* vault-k8s
* CI dependencies

## 0.22.1 (October 26th, 2022)

Changes:
* `vault` updated to 1.12.0 [GH-803](https://github.com/hashicorp/vault-helm/pull/803)
* `vault-k8s` updated to 1.0.1 [GH-803](https://github.com/hashicorp/vault-helm/pull/803)

### CODE REVIEW

**Review by swenson — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,9 +13,9 @@ jobs:
 
   chart-verifier:
     docker:
-      - image: docker.mirror.hashicorp.services/cimg/go:1.16
+      - image: docker.mirror.hashicorp.services/cimg/go:1.19.2
     environment:
-      BATS_VERSION: "1.3.0"
+      BATS_VERSION: "1.8.2"
```
Comment (tvoran) [inline .circleci/config.yml:18]: We could remove all the circleci jobs from this file except for `update-helm-charts-index`. I think that's the only one left running in circle.

Comment (tomhjp) [reply .circleci/config.yml]: Oh good point, I didn't notice they've already been removed from the workflows section of the file anyway so they were already unused.

**Comment by tomhjp** [comment]
I'll merge for now as `images-are-certified` is the only failing check, and I expect that to be resolved once we've sorted the issues publishing `vault-k8s` 1.0.1 to the upstream repository. The GitHub release doesn't need to block on that, but publishing the release on the OpenShift repository does.

**Pull request merged**